src/fix/fixCodec.h

```diff
@@ -0,0 +1,156 @@
+#pragma once
+#include <stdint.h>
+#include <iostream>
+#include <map>
+#include "cdr.h"
+#include "utils.h"
+#include "codec.h"
+#include "fixGroup.h"
+#include "fixField.h"
+#include <libxml/parser.h>
+#include <libxml/tree.h>
+
+namespace neueda
+{
+
+using namespace std;
+
+class fixCodec;
+
+typedef map<int64_t, fixField*> fixFieldDefs;
+typedef vector<int64_t> tags;
+typedef vector<fixField*> fields;
+typedef map<int64_t, fixGroup*> groupMap;
+typedef map<string, int64_t> fieldNameToTagMap;
+typedef map<string, string> fieldNameToTypeMap;
+typedef map<char, fields> msgFieldMap;
+
+class fixCodec: public codec
+{
+public:
+    bool loadDataDictionary (const char* xml, string& err);
+
+    codecState decode (cdr& target,
+                       const void* buf,
+                       size_t len,
+                       size_t& used);
+
+    codecState encode (const cdr& source,
+                       void* buf,
+                       size_t len,
+                       size_t& used);
+
+    string getType () const { return "fix"; }
+
+private:
+    codecState encodeCdr (const cdr& d,
+                          void* buf,
+                          size_t len,
+                          size_t& used);
+
+    codecState encodeHeader (const string& beginString,
+                             const size_t& bodyLen,
+                             const string& msgType,
+                             char* p,
+                             const size_t len,
+                             size_t& used);
+
+    codecState decodeTagValue (char*& buf,
+                               size_t len,
+                               size_t& used,
+                               int64_t& tag,
+                               string& val);
+
+    codecState decodeGroup (cdr& d,
+                            fixGroup* group,
+                            char*& buf,
+                            size_t len,
+                            size_t& used,
+                            int64_t& lastTag,
+                            string& lastVal);
+
+    codecState validate (const void* buf,
+                         size_t len,
+                         size_t& used,
+                         string& err);
+
+    void parseFieldType (int64_t tag, string& type);
+
+    bool processDictionaryFields (xmlNode* node, string& err);
+
+    bool processDictionaryGroup (xmlNode* node, string& err);
+
+    bool processDictionaryMessages (xmlNode* node, string& err);
+
+    bool getFixFieldFromNode (xmlNode* node, fixField& f);
+
+    bool getFixTagByName (const string& name, int64_t& tag)
```

Comment:
can you use const_iterator on the members here and make accessors const methods?

Reply:
likewise with all of the getters below too